### PR TITLE
[6.x] Add Ruby 3.0 and Rails 6.1 compatibility

### PR DESCRIPTION
**Pull request by tagliala**
Includes changes in #191

Rails 6.1 needs the following changes:
- Add `owner: nil` to `define_method_attribute` calls
- Pass `options` to error `add` method as keyword argument

Ruby 3 needs the following changes:
- Double splat values arguments passed to `DateTimeValue`

Ref:
- https://api.rubyonrails.org/v6.0.0/classes/ActiveModel/Errors.html#method-i-add
- https://api.rubyonrails.org/v6.1.0/classes/ActiveModel/Errors.html#method-i-add

Fix: #193

### CODE REVIEW

**Comment by tagliala** [comment]
Everything is green, there are interesting failures with ruby-edge

https://travis-ci.org/github/adzap/validates_timeliness/jobs/750986791

I cannot reproduce them with Ruby 3.0.0 on my dev machine

Also, Travis does not have Ruby 3.0.0-rc1

**Comment by tagliala** [comment]
Working also on Ruby 3.0 compatibility

**Comment by tagliala** [comment]
FWIW, this gem is still compatible with Ruby 2.4

**Comment by tagliala** [comment]
Not yet, since the Readme needs to be updated, I'm going to integrate #185

**Comment by tagliala** [comment]
Preview of the new readme: https://github.com/tagliala/validates_timeliness/tree/rails-61#validatestimeliness-

**Comment by ThomasTrzos** [comment]
When are you planning to merge this pull request?

**Comment by slomagnieszka** [comment]
Do you have a plan to merge it?

**Comment by itay-grudev** [comment]
@adzap This MR seems to cover all the checks to get merged.

**Comment by tagliala** [comment]
Not yet at the moment, it needs some updates

**Comment by pedrofurtado** [comment]
@adzap Any news about this issue? Happy to see this merged and released asap 🎉 🍻

**Comment by StoneGod** [comment]
Hi, what about released?

**Comment by adzap** [comment]
@tagliala thanks for getting this together. But I will need a few changes. Please remove the Rdoc conversion, which we can look at that later.

Also, this will be for v6 of the gem which will be for Rails v6.x only, so the v5 appraisals and travis config should be removed.

File: README.rdoc [hunk 1]
```diff
@@ -30,13 +34,13 @@ If you a looking for the old version for Rails 4.x go here [https://github.com/a
 == Installation
 
   # in Gemfile
-  gem 'validates_timeliness', '~> 5.0.0.beta1'
+  gem 'validates_timeliness', '~> 6.0.0.alpha1'
```
Comment (tagliala) [inline README.rdoc:37]: opinionated

File: README.rdoc [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
-Complete validation of dates, times and datetimes for Rails 5.x and ActiveModel.
+Complete validation of dates, times and datetimes for Rails 6.x and ActiveModel.
 
-If you a looking for the old version for Rails 4.x go here [https://github.com/adzap/validates_timeliness/tree/4-0-stable].
+Old Rails versions:
+
+* Rails 4.x: [https://github.com/adzap/validates_timeliness/tree/4-0-stable]
+
+* Rails 5.x: [https://github.com/adzap/validates_timeliness/tree/5-0-stable]
```
Comment (tagliala) [inline README.rdoc:14]: opinionated, feel free to suggest a better approach

File: Rakefile [hunk 3]
```diff
@@ -19,10 +19,11 @@ end
 
 desc 'Generate documentation for plugin.'
 Rake::RDocTask.new(:rdoc) do |rdoc|
+  rdoc.main = 'README.rdoc'
   rdoc.rdoc_dir = 'rdoc'
   rdoc.title    = 'ValidatesTimeliness'
-  rdoc.options << '--line-numbers' << '--inline-source'
-  rdoc.rdoc_files.include('README')
+  rdoc.options << '--line-numbers'
+  rdoc.rdoc_files.include('README.rdoc')
```
Comment (tagliala) [inline Rakefile:26]: README is not a file

File: Rakefile [hunk 4]
```diff
@@ -19,10 +19,11 @@ end
 
 desc 'Generate documentation for plugin.'
 Rake::RDocTask.new(:rdoc) do |rdoc|
+  rdoc.main = 'README.rdoc'
   rdoc.rdoc_dir = 'rdoc'
   rdoc.title    = 'ValidatesTimeliness'
-  rdoc.options << '--line-numbers' << '--inline-source'
```
Comment (tagliala) [inline Rakefile:24]:
```
$ rake rdoc
option --inline-source is deprecated: source code is now always inlined
```

File: Rakefile [hunk 5]
```diff
@@ -19,10 +19,11 @@ end
 
 desc 'Generate documentation for plugin.'
 Rake::RDocTask.new(:rdoc) do |rdoc|
+  rdoc.main = 'README.rdoc'
```
Comment (tagliala) [inline Rakefile:22]: Uses README as main page.

### Before:

image

### After:

image

File: LICENSE [hunk 6]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2008-2010 Adam Meehan
+Copyright (c) 2008-2021 Adam Meehan
```
Comment (tagliala) [inline LICENSE:1]: +11

File: lib/validates_timeliness/version.rb [hunk 7]
```diff
@@ -1,3 +1,3 @@
 module ValidatesTimeliness
-  VERSION = '5.0.0'
+  VERSION = '6.0.0.alpha1'
```
Comment (tagliala) [inline lib/validates_timeliness/version.rb:2]: opinionated

File: README.rdoc (hunk 2 above)
Comment (adzap) [reply README.rdoc:14]: this is food

File: Rakefile (hunk 5 above)
Comment (adzap) [reply Rakefile:22]: Nice work!

**Comment by adzap** [comment]
@tagliala excellent work! so many little tweaks and fixes in the README because of the markdown conversion.

**Comment by pedrofurtado** [comment]
@adzap Happy to see this released in official version for us 🍻 🎉

**Comment by adzap** [comment]
Looking for feedback from those using the alpha release. A number of you were quite eager for it, so in order to get a final release I would like some confirmation of successful usage.

**Comment by itay-grudev** [comment]
Can we add automatic testing to the repository in addition to the automatic builds?

**Comment by adzap** [comment]
We already do via Travis.

**Comment by itay-grudev** [comment]
It's just called build and I thought it's for gem build and publish instead
of testing.

**Comment by adzap** [comment]
What's it called in other repos? Can you give me an example of some others that are perhaps more up to date?

**Pull request merged**